sources/packages/backend/apps/api/src/main.ts

```diff
@@ -24,6 +25,24 @@ async function bootstrap() {
   // Setting global prefix
   app.setGlobalPrefix("api");
 
+  // Using helmet
+  app.use(
+    helmet({
+      strictTransportSecurity: {
+        maxAge: 31536000,
+        includeSubDomains: true,
+        preload: true,
+      },
+    }),
+  );
+
+  // Adding headers not covered by helmet
+  app.use((_, res, next) => {
+    res.setHeader("Cache-Control", "no-store");
```

Comment:
If we would like to enable cache for a particular API endpoint, `config` for instance, with the configuration at this level would it still be possible?

Reply:
Yes, that would be something like this:
```js
  app.use((req, res, next) => {
    if (req.url.contains("/api/config")) {
      res.setHeader("Cache-Control", "public, max-age=3600");
      res.setHeader("Pragma", "public");
    } else {
      res.setHeader("Cache-Control", "no-store");
      res.setHeader("Pragma", "no-cache");
    }
    next();
  });
```

Reply:
Wava scan recommends to prevent caching for the `config` endpoint though.
![image](https://github.com/bcgov/SIMS/assets/78114138/c147b893-804c-4ac0-befa-c07b140ffdf6)
At least it was not clear if just having any values for "Cache-Control" and "Pragma" headers would satisfy that issue.

Reply:
Maybe I was not clear when talking about adding a cache to an endpoint I was referring to the below.
https://docs.nestjs.com/controllers#headers
```ts
@Get()
@Header('Cache-Control', 'none')
getConfig() {
  return { ourConfig };
}
```

Reply:
Yes, that can be achieved that way too if needed and I did a test here to confirm that the `@Header()` decorator will be applied even if we have a middleware function setting the same header globally.